src/main/java/com/develhope/spring/deals/services/OrderService.java

```diff
@@ -0,0 +1,52 @@
+package com.develhope.spring.deals.services;
+
+import com.develhope.spring.deals.models.Order;
+import com.develhope.spring.deals.models.OrderStatus;
+import com.develhope.spring.deals.repositories.OrderRepository;
+import com.develhope.spring.users.models.Roles;
+import com.develhope.spring.users.models.User;
+import com.develhope.spring.users.repositories.UserRepository;
+import com.develhope.spring.vehicles.models.Vehicle;
+import com.develhope.spring.vehicles.repositories.VehicleRepository;
+import com.develhope.spring.vehicles.services.VehicleService;
+import com.develhope.spring.vehicles.vehicleEnums.MarketStatus;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class OrderService {
+
+    @Autowired
+    private VehicleRepository vehicleRepository;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private VehicleService vehicleService;
+
+    @Autowired
+    private OrderRepository orderRepository;
+
+    public Order createOrder(boolean downPayment, long vehicleId, User user) throws Exception {

```

Comment:
devi passare Order order anche, altrimenti come fai a crearlo? In realtà ancora meglio secondo quanto poi detto sotto dovresti passare OrderDTO orderDTO